presto-main/src/main/resources/webapp/src/components/QueryList.jsx

```diff
@@ -62,14 +62,26 @@ export class QueryListItem extends React.Component {
             }
         }
 
+        const maxLines = 7;
+        const maxWidthPerLine = 75;
+        let lineCount = 0;
         let formattedQueryText = "";
 
         for (let i = 0; i < lines.length; i++) {
             const trimmedLine = lines[i].substring(minLeadingWhitespace).replace(/\s+$/g, '');
+            lineCount += Math.ceil(trimmedLine.length / maxWidthPerLine);
 
             if (trimmedLine.length > 0) {
                 formattedQueryText += trimmedLine;
 
+                if (formattedQueryText.length >= 300) {
```

Comment:
This needs to be > 300 instead of >= 300.
Otherwise, when formattedQueryText.length == 300, truncateString in line 91 will return formattedQueryText without the '...' which is not correct as there could be more line which are appended to formattedQueryText yet.

Reply:
done